src/common/serial_interface.cc

```diff
@@ -90,54 +85,41 @@ SerialInterface::setAttributes(int speed)
     tty.c_cc[VTIME] = 5;
 
     if (tcsetattr(m_Serial_fd, TCSANOW, &tty) != 0) {
-        std::cerr << "Error in setup from tcgetattr:" << strerror(errno) << std::endl;
-        return false;
+        throw std::runtime_error("Error in setup from tcgetattr:" + std::string(strerror(errno)));
     }
-    return true;
 }
 
-bool
+void
 SerialInterface::setBlocking(bool should_block)
 {
     struct termios tty;
     memset(&tty, 0, sizeof tty);
     if (tcgetattr(m_Serial_fd, &tty) != 0) {
-        std::cerr << "Error in setup from tcgetattr:" << strerror(errno) << std::endl;
-        return false;
+        throw std::runtime_error("Error in setup from tcgetattr:" + std::string(strerror(errno)));
     }
 
     tty.c_cc[VMIN] = should_block ? 1 : 0;
     tty.c_cc[VTIME] = 5; // 0.5 seconds read timeout
 
     if (tcsetattr(m_Serial_fd, TCSANOW, &tty) != 0) {
-        std::cerr << "Error in setup setting term attributes:" << strerror(errno) << std::endl;
-        return false;
+        throw std::runtime_error("Error in setup setting term attributes:" + std::string(strerror(errno)));
     }
-    return true;
 }
 
-bool
+void
 SerialInterface::readByte(uint8_t &byte)
 {
-    char data; // = i2c_smbus_read_byte(m_I2C);
-    ::read(m_Serial_fd, &data, 1);
-    if (data != 1) {
-        std::cerr << "Failed to read byte from Serial port" << std::endl;
-        return false;
-    } else {
-        byte = (uint8_t) data;
-        return true;
+    const ssize_t ret = ::read(m_Serial_fd, reinterpret_cast<char *>(&byte), 1);
+    if (ret < 0) {
+        throw std::runtime_error("Failed to read byte from serial port");
```

Comment:
but now, if it's in non-blocking mode, how does it signal whether it's read a byte or not?

Reply:
Hmm good point... I guess it should return ``false`` if there's no data...

Reply:
:+1: